types/link_test.go

```diff
@@ -101,10 +101,82 @@ func TestUnmarshalRawLinksYaml(t *testing.T) {
 			wantErr: false,
 			want: LinkDefinition{
 				Type: string(LinkTypeBrief),
-				LinkConfig: LinkConfig{
-					Endpoints: []string{
-						"srl1:e1-5",
-						"srl2:e1-5",
+				Link: &LinkVEthRaw{
+					Endpoints: []*EndpointRaw{
+						NewEndpointRaw("srl1", "e1-5", ""),
+						NewEndpointRaw("srl2", "e1-5", ""),
+					},
+				},
+			},
+		},
+		{
+			name: "brief link with macvlan endpoint",
+			args: args{
+				yaml: []byte(`
+                    endpoints: ["srl1:e1-1", "macvlan:eth0"]`,
+				),
+			},
+			wantErr: false,
+			want: LinkDefinition{
+				Type: string(LinkTypeBrief),
+				Link: &LinkMacVlanRaw{
+					HostInterface: "eth0",
+					Endpoint:      NewEndpointRaw("srl1", "e1-1", ""),
+				},
+			},
+		},
+		{
+			name: "brief link with mgmt-net endpoint",
+			args: args{
+				yaml: []byte(`
+                    endpoints: ["srl1:e1-1", "mgmt-net:srl1-e1-1"]`,
+				),
+			},
+			wantErr: false,
+			want: LinkDefinition{
+				Type: string(LinkTypeBrief),
+				Link: &LinkMgmtNetRaw{
+					HostInterface: "srl1-e1-1",
+					Endpoint:      NewEndpointRaw("srl1", "e1-1", ""),
+				},
+			},
+		},
+		{
+			name: "brief link with host endpoint",
+			args: args{
+				yaml: []byte(`
+                    endpoints: ["srl1:e1-1", "host:srl1-e1-1"]`,
+				),
+			},
+			wantErr: false,
+			want: LinkDefinition{
+				Type: string(LinkTypeBrief),
+				Link: &LinkHostRaw{
+					HostInterface: "srl1-e1-1",
+					Endpoint:      NewEndpointRaw("srl1", "e1-1", ""),
+				},
+			},
+		},
+		{
+			name: "veth link",
+			args: args{
+				yaml: []byte(`
+                    type:              veth
+                    endpoints:
+                      - node:          srl1
+                        interface:     e1-1
+                        mtu:           1400
+                      - node:          srl2
+                        interface:     e1-2
+                `),
+			},
```

Comment:
@steiler I've added `mtu` here to check what you said in https://github.com/srl-labs/containerlab/pull/1475#issuecomment-1666545946
locally I didn't encounter an error.

Reply:
ha, but the test is incomplete, I have to add MTU to the expected link and it is not set there. Will check what's up

Reply:
should be all good
[76fb628](https://github.com/srl-labs/containerlab/pull/1475/commits/76fb628d54ee91f0c42591dad0f4cf0a80c4d46b)

initially put MTU on the wrong object.